src/wp-includes/class-wp-theme.php

```diff
@@ -1993,16 +2002,31 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/** This filter is documented in wp-includes/theme.php */
+		$cache_expiration = (int) apply_filters( 'wp_cache_themes_persistently', self::$cache_expiration, 'theme_block_patterns' );
```

Comment:
We can't reuse that filter here, as it does not behave the same. Additionally, the filter name does not make sense for a filter that controls the cache TTL. The current name is more applicable for a "boolean" decision whether or not to cache, but that's not what we're doing here.

How about `wp_theme_files_cache_ttl` for the new filter's name?